src/scimPatch.ts

```diff
@@ -247,9 +247,21 @@ function navigate(inputSchema: any, paths: string[]): any {
  */
 function addOrReplaceAttribute(property: any, patch: ScimPatchAddReplaceOperation): any {
     if (Array.isArray(property)) {
-        const isReplace = patch.op.toLowerCase() === 'replace';
-        if(isReplace && Array.isArray(patch.value))
-            return patch.value;
+        console.log(property)
+
+        if(Array.isArray(patch.value)) {
+            const op = patch.op.toLowerCase();
+            const isReplace = op === "replace";
+            if (isReplace)
+                return patch.value;
+            const isAdd = op === "add";
+            if (isAdd) {
+                if (!property.includes(patch.value)) {
+                    return property.concat(patch.value);
+                }
+            }                
```

Comment:
We can have only one if statement here, it will better for readibility.
```suggestion
            if (isAdd && !property.includes(patch.value))
                return property.concat(patch.value);
```